packages/core/src/components/panel-stack2/panelTypes.ts

```diff
@@ -21,7 +21,7 @@ export interface Panel<P> {
     /**
      * The renderer for this panel.
      */
-    renderPanel: (props: PanelProps<P>) => JSX.Element | null;
+    renderPanel: (props: PanelProps<P>) => ReturnType<React.FC<PanelProps<P>>>;
```

Comment:
I think this could also just use `React.FC` entirely.

```suggestion
    renderPanel: React.FC<PanelProps<P>>;
```

But that is a bit more strict than what we had before this PR since `React.FC` has additional optional properties on it.

```ts
interface FunctionComponent<P = {}> {
    (props: P, context?: any): ReactNode;
    propTypes?: WeakValidationMap<P> | undefined;
    contextTypes?: ValidationMap<any> | undefined;
    defaultProps?: Partial<P> | undefined;
    displayName?: string | undefined;
}
```

Reply:
I tried this out and it seems to work fine with both functions that are annotated as `React.FC` and plain (un-annotated) ones which return any of the `ReactNode` types: https://codesandbox.io/p/sandbox/react-18-panelstack2-types-debugging-typescript-v5-x-hz29zl?file=%2Fsrc%2FCoreExample.tsx%3A18%2C1

so I think we can go ahead with this suggested change

Reply:
Nice. Thank you for the help testing that `React.FC<...>` works for those cases. PR should be updated for this now.